rfqservice/oracle.go

```diff
@@ -0,0 +1,118 @@
+package rfqservice
+
+import (
+	"context"
+	"crypto/tls"
+	"net/url"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// OracleAskingPriceResponse is a struct that holds the price oracle's suggested
+// asking price for an asset.
+type OracleAskingPriceResponse struct {
+	// AskingPrice is the asking price of the quote.
+	AskingPrice *lnwire.MilliSatoshi
+
+	// Expiry is the asking price expiry lifetime unix timestamp.
+	Expiry uint64
+
+	// Err is the error returned by the price oracle service.
+	Err *rfqmsg.RejectErr
+}
+
+// PriceOracle is an interface that provides exchange rate information for
+// assets.
+type PriceOracle interface {
+	// QueryAskingPrice returns the asking price for the given asset amount.
+	QueryAskingPrice(ctx context.Context, assetId *asset.ID,
```

Comment:
Can combine asset+group key into single identifier, see above.

Reply:
Will do :+1:

Reply:
I'll get that done in a future PR.